crates/ibc/src/core/ics04_channel/handler/timeout.rs

```diff
@@ -306,9 +292,66 @@ mod tests {
     }
 
     #[rstest]
-    #[ignore = "implement and make clear that the timeout is indeed not reached"]
-    fn timeout_fail_proof_timeout_not_reached(_fixture: Fixture) {
-        // TODO
+    fn timeout_fail_proof_timeout_not_reached(fixture: Fixture) {
+        let Fixture {
+            context,
+            mut msg,
+            chan_end_on_a_unordered,
+            conn_end_on_a,
+            client_height,
+            ..
+        } = fixture;
+
+        // timeout timestamp has not reached yet
+        let timeout_timestamp_on_b =
+            (msg.packet.timeout_timestamp_on_b + core::time::Duration::new(10, 0)).unwrap();
+        msg.packet.timeout_timestamp_on_b = timeout_timestamp_on_b;
+        let packet_commitment = compute_packet_commitment(
+            &msg.packet.data,
+            &msg.packet.timeout_height_on_b,
+            &msg.packet.timeout_timestamp_on_b,
+        );
+
+        let packet = msg.packet.clone();
+
+        let mut context = context
+            .with_client(&ClientId::default(), client_height)
+            .with_connection(ConnectionId::default(), conn_end_on_a)
+            .with_channel(
+                PortId::default(),
+                ChannelId::default(),
+                chan_end_on_a_unordered,
+            )
+            .with_packet_commitment(
+                packet.port_id_on_a,
+                packet.chan_id_on_a,
+                packet.seq_on_a,
+                packet_commitment,
+            );
+
+        context
+            .store_update_time(
+                ClientId::default(),
+                client_height,
+                Timestamp::from_nanoseconds(5).unwrap(),
+            )
+            .unwrap();
+        context
+            .store_update_height(
+                ClientId::default(),
+                client_height,
+                Height::new(0, 4).unwrap(),
+            )
+            .unwrap();
+
+        let res = validate(&context, &msg);
+
+        assert!(
+            res.unwrap_err()
+                .to_string()
+                .starts_with("ICS04 Packet error: Packet timeout"),
```

Comment:
As inconvenient as it is, can we match on the error instead of matching on the string representation?

(Another good example why our current errors are difficult to use: #270).

Reply:
or just `is_err()` if it's too unwieldy